chat-plugins/uno.js

```diff
@@ -0,0 +1,694 @@
+function shuffleDeck(dArray) {
+	let tDeck = [];
+	let reiterations = dArray.length;
+	for (let i = 0; i < reiterations; i++) {
+		let rand = ~~(dArray.length * Math.random());
+		tDeck.push(dArray[rand]);
+		dArray.splice(rand, 1);
+	}
+	return tDeck;
+}
+
+function getCardName(id, colour) {
+	let type = {
+		"R": "Red",
+		"Y": "Yellow",
+		"B": "Blue",
+		"G": "Green",
+		"Z": "Wild",
+	};
+	let value = {
+		"1": " 1",
+		"2": " 2",
+		"3": " 3",
+		"4": " 4",
+		"5": " 5",
+		"6": " 6",
+		"7": " 7",
+		"8": " 8",
+		"9": " 9",
+		"0": " 0",
+		"R": " Reverse",
+		"S": " Skip",
+		"+2": " Draw Two",
+		"+4": " Draw Four",
+		"W": "card",
+	};
+	if (colour) return "<font color=\"" + type[id.charAt(0)].toLowerCase().replace("yellow", "orange") + "\">" + type[id.charAt(0)] + value[id.slice(1)] + "</font>";
+	return type[id.charAt(0)] + value[id.slice(1)];
+}
+
+
+function buildGameScreen(uhtmlid, top, roomid, hand, message, change, pass) {
+	let html = (message ? "|uhtmlchange|" + uhtmlid : "|uhtml|" + uhtmlid) + "|";
+	//make top card not clickable
+	let topCard = "<center>" + buildCard(top, true) + "</center>";
+	if (change) {
+		topCard += "<br><center>Colour changed to: <font color=\"" + getColour(change) + "\">" + getColour(change, true) + "</font></center>";
+	}
+	let yourHand = buildHand(hand);
+	message = message ? "<font color=\"red\"><b>" + message + "</b></font>" : "";
+	return html + "<table border=1 style=\"border-collapse: collapse;\"><tr><td>&nbsp;<b>Top Card</b>&nbsp;</td><td>&nbsp;<b>Your Hand</b>&nbsp;</td></tr>" + "<tr><td>" + topCard + "</td><td>" + yourHand + "</td></tr><tr><td>" + (pass ? passButton : drawButton) + "</td><td>" + message + "</td></tr></table>";
+}
+
+//for wildcards that summon a colour change
+function getColourChange(buffer, hand, uhtmlid) {
+	return "|uhtmlchange|" + uhtmlid + "|<table border=1 style=\"border-collapse: collapse;\"><tr><td>Choose which colour to change to:<br>" + '<button style="background: red; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' R"><font color="white" size=4>Red</font></button></center>' + '<button style="background: yellow; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' Y"><font color="black" size=4>Yellow</font></button></center>' + '<button style="background: blue; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' B"><font color="white" size=4>Blue</font></button></center>' + '<button style="background: green; border: 2px solid rgba(33 , 68 , 72 , 0.59) ; width: 80px ; border-radius: 5px , auto" name="send" value="' + buffer + ' G"><font color="white" size=4>Green</font></button></td></tr>' + '<tr><td>Your Cards: <br>' + buildHand(hand, true) + '</td></tr></table>';
+}
+
+function getUserName(id) {
+	return Users(id) ? Users(id).name : id;
+}
+
+class Game {
```

Comment:
He's referring to the new <a href="https://github.com/Zarel/Pokemon-Showdown/blob/master/room-game.js">RoomGame API</a>.